test/e2e/networking.go

```diff
@@ -241,6 +208,50 @@ var _ = Describe("Networking", func() {
 		Expect(string(body)).To(Equal("pass"))
 	})
 
+	// Marked with [Skipped] to skip the tests by default.  These
+	// tests provide more granular failure modes than the intra-pod
+	// test and are intended to be used by developers of network
+	// plugins.
+	Describe("[Skipped] Granular Checks", func() {
+
+		It("should function for pod communication on a single node", func() {
+
+			By("Picking a node")
+			nodes := getReadyNodes(f)
+			node := nodes.Items[0]
+
+			By("Creating a webserver pod")
+			podName := "same-node-webserver"
+			defer f.Client.Pods(f.Namespace.Name).Delete(podName, nil)
+			ip := launchWebserverPod(f, podName, node.Name)
+
+			By("Checking that the webserver is accessible from a pod on the same node")
+			expectNoError(checkConnectivityToHost(f, node.Name, "same-node-wget", ip))
+		})
+
+		It("should function for pod communication between nodes", func() {
+
+			podClient := f.Client.Pods(f.Namespace.Name)
+
+			By("Picking multiple nodes")
+			nodes := getReadyNodes(f)
+
+			if len(nodes.Items) == 1 {
```

Comment:
This test is intended to verify pod connectivity across nodes and I think having only one node effectively invalidates the test.  I'll add it if you insist, though, as I can't claim to be an expert on kube testing convention.